src/piece_state/mod.rs

```diff
@@ -0,0 +1,15 @@
+use serde::{Deserialize, Serialize};
+use serde_json::Result;
+
+#[derive(Serialize, Deserialize, Debug)]
```

Comment:
Since this syntax is a bit odd, might be nice to comment here about what it does.